qiskit_experiments/library/hamiltonian/heat_analysis.py

```diff
@@ -0,0 +1,143 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Analysis for HEAT experiments.
+"""
+
+from typing import List
+
+import numpy as np
+
+from qiskit_experiments.curve_analysis import ErrorAmplificationAnalysis
+from qiskit_experiments.exceptions import AnalysisError
+from qiskit_experiments.framework import (
+    CompositeAnalysis,
+    ExperimentData,
+    AnalysisResultData,
+    Options,
+    FitVal,
+)
+
+
+class HeatElementAnalysis(ErrorAmplificationAnalysis):
+    """An analysis class for HEAT experiment to define the fixed parameters.
+
+    # section: overview
+
+        This is standard error amplification analysis.
+
+    # section: see_also
+        qiskit_experiments.curve_analysis.ErrorAmplificationAnalysis
+    """
+
+    __fixed_parameters__ = ["angle_per_gate", "phase_offset", "amp"]
+
+    @classmethod
+    def _default_options(cls) -> Options:
+        """Default analysis options."""
+        options = super()._default_options()
+        options.angle_per_gate = np.pi
+        options.phase_offset = np.pi / 2
+        options.amp = 1.0
+
+        return options
+
+
+class HeatAnalysis(CompositeAnalysis):
+    r"""A composite error amplification analysis to get unitary error coefficients.
```

Comment:
The model is written in the ``ZXHeat`` class rather than here since this is not restricted to specific Hamiltonian and it might be hard to write general model.

Reply:
Why not write an example model and make it clear that this has a broader use.

Reply:
Any example of the model?